cli/schedule.go

```diff
@@ -0,0 +1,298 @@
+
+	return scheduleCmd
+}
+
+func scheduleShow() *cobra.Command {
+	showCmd := &cobra.Command{
+		Annotations: workspaceCommand,
+		Use:         "show <workspace_name>",
+		Short:       "Show workspace schedule",
+		Long:        scheduleShowDescriptionLong,
+		Args:        cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := createClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			workspace, err := namedWorkspace(cmd, client, args[0])
+			if err != nil {
+				return err
+			}
+
+			return displaySchedule(workspace, cmd.OutOrStdout())
+		},
+	}
+	return showCmd
+}
+
+func scheduleStart() *cobra.Command {
+	cmd := &cobra.Command{
+		Annotations: workspaceCommand,
+		Use:         "start <workspace_name> { <start-time> [day-of-week] [location] | manual }",
+		Example:     `start my-workspace 9:30AM Mon-Fri Europe/Dublin`,
+		Short:       "Edit workspace start schedule",
+		Long:        scheduleStartDescriptionLong,
+		Args:        cobra.RangeArgs(2, 4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := createClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			workspace, err := namedWorkspace(cmd, client, args[0])
+			if err != nil {
+				return err
+			}
+
+			var schedStr *string
+			if args[1] != "manual" {
+				sched, err := parseCLISchedule(args[1:]...)
+				if err != nil {
+					return err
+				}
+
+				schedStr = ptr.Ref(sched.String())
+			}
+
+			err = client.UpdateWorkspaceAutostart(cmd.Context(), workspace.ID, codersdk.UpdateWorkspaceAutostartRequest{
+				Schedule: schedStr,
+			})
+			if err != nil {
+				return err
+			}
+
+			updated, err := namedWorkspace(cmd, client, args[0])
+			if err != nil {
+				return err
+			}
+			return displaySchedule(updated, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+func scheduleStop() *cobra.Command {
+	return &cobra.Command{
+		Annotations: workspaceCommand,
+		Args:        cobra.ExactArgs(2),
+		Use:         "stop <workspace_name> { <duration> | manual }",
+		Example:     `stop my-workspace 2h30m`,
+		Short:       "Edit workspace stop schedule",
+		Long:        scheduleStopDescriptionLong,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := createClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			workspace, err := namedWorkspace(cmd, client, args[0])
+			if err != nil {
+				return err
+			}
+
+			var durMillis *int64
+			if args[1] != "manual" {
+				dur, err := parseDuration(args[1])
+				if err != nil {
+					return err
+				}
+				durMillis = ptr.Ref(dur.Milliseconds())
+			}
+
+			if err := client.UpdateWorkspaceTTL(cmd.Context(), workspace.ID, codersdk.UpdateWorkspaceTTLRequest{
+				TTLMillis: durMillis,
+			}); err != nil {
+				return err
+			}
+
+			updated, err := namedWorkspace(cmd, client, args[0])
+			if err != nil {
+				return err
+			}
+			return displaySchedule(updated, cmd.OutOrStdout())
+		},
+	}
+}
+
+func scheduleOverride() *cobra.Command {
```

Comment:
Honestly `bump-stop` or `delay-stop` could work well. I was wrong about the need for a directionally neutral term since the duration is applied additively by default 🤦🏽‍♂️. As long as they can do negatives we have a good UX.

Reply:
I'd honestly steer clear of `bump` after the changes in https://github.com/coder/coder/pull/2230 changing the semantics of the CLI command from `add duration to current deadline` (non-idempotent) to `set deadline to now plus duration` (idempotent). 

`override-stop` or `override stop` makes sense for the current semantics.

An alternative would be to add a flag `--update-running` to the `schedule stop` command but that's probalby too much coupling.

Reply:
Gotcha, amongst those options I think `override-stop` is most clear. `override` with one nested command would be odd.